### PR TITLE
#484 DevEx - Create a Dev Container 

### LINKED ISSUE #484: 🧑‍💻 DevEx - Create a Dev Container
Create a Dev container for the project as per the https://www.ssw.com.au/rules/dev-containers/ Rule

### PR DESCRIPTION
**Pull request by johnxu16**
Add devcontainer configuration based on https://ssw.com.au/rules/dev-containers

Env:
Windows 11 (10.0.22621.1555)
Docker Desktop 4.19.0 (106363)
WSL2 1.2.5.0!

✅ Works on MacOS 

⚠️ Issues on Windows:
## Try 1 (init devcontainer inside WSL2 ubuntu 22.04.2 LTS )
- [x] image build
- [x] yarn install
- [x] extensions
- [x] port forwarding 

## Try 2 (init devcontainer on windows 11)
- [x] image build
- [x] yarn install
- [x] extensions
- [x] port forwarding

MicrosoftTeams-image
**Figure: Yarn failed to install**

image
**Figure: EIO: i/o error**

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ARG VARIANT="16-bullseye"
+FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:${VARIANT}
+
+ARG NODE_VERSION="20"
+RUN umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} && corepack enable
+
+# RUN apt-get clean -y \
+# && rm -rf /var/lib/apt/lists/* /tmp/library-scripts \
+# && apt-get update \
+# && apt-get -y install --no-install-recommends git
```
Comment (bradystroud) [inline .devcontainer/Dockerfile]: Remove these

**Review by bradystroud — CHANGES_REQUESTED** [review]
As per our conversation, please add the new steps to the readme.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (johnxu16) [reply .devcontainer/Dockerfile]: Done - Removed 🙂

**Comment by johnxu16** [comment]
> ## Try 2 (init devcontainer on windows 11)
> * [x]  image build
> * [x]  yarn install
> * [x]  extensions
> * [x]  port forwarding
> 
> MicrosoftTeams-image **Figure: Yarn failed to install**
> 
> image **Figure: EIO: i/o error**

After freeing up some space in c drive, problems solved

**Review by bradystroud — APPROVED** [review]

**Pull request merged**